### PR TITLE
task: use simple display names for config sources

### LINKED ISSUE #24192: Refine how ConfigSource names are being used
### Description

With #13608 it was noted that the direct exposure of ConfigSource names in show-config and now the additional property validation that is being done is not a good user experience.  At this level users should probably see something simple like - cli, env, just the properties file name (conf/keycloak.conf)

### PR DESCRIPTION
**Pull request by shawkins**
closes: #24192

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java [hunk 1]
```diff
@@ -76,14 +76,14 @@ void testFailRuntimeOptions(LaunchResult result) {
...
-        cliResult.assertError("Invalid value for option 'kc.db': invalid. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres. From ConfigSource KcEnvVarConfigSource");
+        cliResult.assertError("Invalid value for option 'KC_DB': invalid. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres");
     }
 
     @Test
     @RawDistOnly(reason = "Raw is enough and we avoid issues with including custom conf file in the container")
     public void testFailInvalidOptionInConf(KeycloakDistribution distribution) {
         CLIResult cliResult = distribution.run(CONFIG_FILE_LONG_NAME + "=" + Paths.get("src/test/resources/BuildCommandDistTest/keycloak.conf").toAbsolutePath().normalize(), "build");
-        cliResult.assertError("Invalid value for option 'kc.db': foo. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres. From ConfigSource PropertiesConfigSource[source");
+        cliResult.assertError("Invalid value for option 'kc.db' in configuration file: foo. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres");
```
Comment (mabartos) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:86]: It'd be great to know what exactly is the configuration file. Is it feasible? WDYT?

**Review by mabartos — COMMENTED** [review]
@shawkins I've just made a quick look and LGTM. Just one thing that might be considered for better UX.

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java (hunk 1 above)
Comment (shawkins) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java:86]: Possibly. With the way the logic is currently there can be only one configuration file - keycloak.conf or whatever the user provides to --config-file. It seemed good enough to refer to it abstractly.

Alternatively we can try to extract the short name, or even the full path (but I wasn't sure we even wanted full file paths in these messages).

Comment (shawkins) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java]: It will now show the filename instead.

Comment (mabartos) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java]: nice, thanks!

**Review by mabartos — DISMISSED** [review]
@shawkins Nice stuff; I like it now. Thanks!

I don't see any issue with this approach even for the log message with the admin-inserted data.

**Comment by mabartos** [comment]
@shawkins Could you please resolve conflicts?

**Review by mabartos — APPROVED** [review]

**Comment by mabartos** [comment]
@vmuzikar Do you want to review it as well?

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks!

**Pull request merged**